firewood/src/merkle.rs

```diff
@@ -1465,58 +1341,81 @@ mod test {
                 .expect("CompactSpace init fail");
 
         let store = Box::new(space);
-        let merkle = Merkle::new(store);
+        Merkle::new(store)
+    }
 
-        {
-            let chd = Node::new(NodeType::Leaf(LeafNode(
-                PartialPath(vec![0x1, 0x2, 0x3]),
-                Data(vec![0x4, 0x5]),
-            )));
-            let chd_ref = merkle.put_node(chd.clone()).unwrap();
-            let chd_encoded = chd_ref.get_encoded(merkle.store.as_ref());
-            let new_chd = Node::new(NodeType::decode(chd_encoded).unwrap());
-            let new_chd_encoded = new_chd.get_encoded(merkle.store.as_ref());
-            assert_eq!(chd_encoded, new_chd_encoded);
-
-            let mut chd_encoded: [Option<Vec<u8>>; NBRANCH] = Default::default();
-            chd_encoded[0] = Some(new_chd_encoded.to_vec());
-            let node = Node::new(NodeType::Branch(BranchNode {
-                children: [None; NBRANCH],
-                value: Some(Data("value1".as_bytes().to_vec())),
-                children_encoded: chd_encoded,
-            }));
-
-            let node_ref = merkle.put_node(node.clone()).unwrap();
-
-            let r = node_ref.get_encoded(merkle.store.as_ref());
-            let new_node = Node::new(NodeType::decode(r).unwrap());
-            let new_encoded = new_node.get_encoded(merkle.store.as_ref());
-            assert_eq!(r, new_encoded);
+    fn branch(value: Vec<u8>, encoded_child: Option<Vec<u8>>) -> Node {
+        let children = Default::default();
+        let value = Some(value).map(Data);
+        let mut children_encoded = <[Option<Vec<u8>>; NBRANCH]>::default();
+
+        if let Some(child) = encoded_child {
+            children_encoded[0] = Some(child);
```

Comment:
I think the former is easier to follow. If this weren't inside test-code, I would agree with getting rid of the `mut`. Since this is used for tests though, I'm not concerned about someone coming in and mutating.